src/lib/utils/component.ts

```diff
@@ -12,8 +12,8 @@ export interface Component extends AbstractComponent<ComponentHost> {
 
 }
 
-export interface ComponentClass<T extends Component> extends Function {
-    new(owner: ComponentHost): T;
+export interface ComponentClass<T extends Component, O extends ComponentHost = ComponentHost> extends Function {
```

Comment:
Why can't this type stay `CompilerHost`?

Reply:
Turning on `strictFunctionTypes` requires a stronger type for the `addComponent` method. In particular, the application component in [`application.ts`](https://github.com/TypeStrong/typedoc/blob/master/src/lib/application.ts#L100-L104) adds several components which expect their owner to be of type `Application`. This change makes it possible to specify the owner type in [`component.ts`](https://github.com/TypeStrong/typedoc/pull/840/files#diff-9b7f6610719fb95a47e042dd5d47b905L215) and thus resolves the type incompatibility.